### PR TITLE
prepare for multichain

### PR DESCRIPTION
**Pull request by octobocto**
This PR makes it possible to switch chains internally, and by launching a specific chain with a command line argument/environment variable.

Each chain gets it's own primary color, used for all buttons etc.

I couldn't figure out how to open a new instance of sidesail from within sidesail. To do that, we would need to know where the binary is located, and use OS-native calls to launch a new one. For later! There's nothing stopping the user from launching mulitple instances of sidesail though, and switching chains internally. This is good enough imo.

The only thing missing now is the ethereum rpc, and it's special little view.

### CODE REVIEW

**Comment by octobocto** [comment]
The only thing missing now is the ethereum rpc, and the console view. Console view can be tricky...

**Comment by torkelrogstad** [comment]
This looks very good! Although it appears like we're not switching out the RPC connection when a chain swap occurs. Is that intentionally left for a later PR? Some thoughts on this: 

Not all functionality here will be available for all chains. I'm pretty sure the ETH node doesn't support fetching withdrawal bundles or the BMM loop, for example. So we should make it possible within the chain definitions to mark off functionality that's not available. In that case, the relevant page should be somehow greyed out, explaining that the chain you're using right now doesn't provide this functionality

**Comment by torkelrogstad** [comment]
Think we can use `Platform.resolvedExecutable` from `dart:io` to get the binary, and launch new instances of it

File: lib/config/sidechains.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
+abstract class Sidechain {
+  String name = '';
+  String ticker = '';
```
[the remark this answers is not in the document]
Comment (octobocto) [reply lib/config/sidechains.dart]: Kan kalle den noe annet, men å hardkode SC+nummer overalt er slit. Så jeg gjør det her!

hva er bedre navn enn ticker?

**Comment by octobocto** [comment]
Intentionally left out! But alright, that makes sense.

We need to take the scope down a bit for the SidechainRPC abstrakt class, and pull in a TestchainRPC where needed

**Comment by octobocto** [comment]
What methods are common for all chains?

**Comment by torkelrogstad** [comment]
> What methods are common for all chains?

tbd 😛

**Comment by octobocto** [comment]
> > What methods are common for all chains?
> 
> tbd 😛

Then we have to implement that later, when ethereum rpc is added.

I can fix the general structure on my flight though. New code coming in 8 hours!

**Comment by octobocto** [comment]
> This looks very good! Although it appears like we're not switching out the RPC connection when a chain swap occurs. Is that intentionally left for a later PR? Some thoughts on this:
> 
> Not all functionality here will be available for all chains. I'm pretty sure the ETH node doesn't support fetching withdrawal bundles or the BMM loop, for example. So we should make it possible within the chain definitions to mark off functionality that's not available. In that case, the relevant page should be somehow greyed out, explaining that the chain you're using right now doesn't provide this functionality

Should remove unsupported functionality I think. I’ll make it work!

**Pull request merged**